### ISSUE TITLE
update_kernel.sh --lts-4_4 --bone-rt-channel

### ISSUE THREAD

**Issue by lromor** [RobertCNelson/boot-scripts#44]
Hello, I've installed debian stretch, I wanted to update to the realtime kernel patch using the update_kernel with:

sudo ./update_kernel.sh --lts-4_4 --bone-rt-channel

after the reboot, no leds after the bootloader, dmesg gives me:

```
[   11.087103] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.107135] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.108338] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.109080] leds_gpio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.110075] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.141427] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.171321] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.210816] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.217997] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.248942] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.264872] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.289087] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.315477] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.392396] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.564507] tda998x: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.587196] evdev: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
```

I guess the kernel modules should be recompiled or updated. By searching around it seems an ABI problem, from where could I update the kernel without having this issue?

**Comment by RobertCNelson** [comment]
@lromor this is fallout from gcc6 in stretch..

**Comment by lromor** [comment]
Sorry to bother you @RobertCNelson , but can I ask you what does that mean?

**Comment by RobertCNelson** [comment]
@lromor with gcc-6 (6.2.0-7) --enable-default-pie  was switched on.

Everyone is still dealing with fallout from that..

```
gcc-6 (6.2.0-7) unstable; urgency=medium

  * Update to SVN 20161018 (r241301, 6.2.1) from the gcc-6-branch.
    - Fix PR libstdc++/77987, PR libstdc++/77322, PR libstdc++/72820,
      PR libstdc++/77994, PR tree-optimization/77937, PR c++/71912,
      PR tree-optimization/77937, PR tree-optimization/77943,
      PR bootstrap/77995, PR fortran/77978, PR fortran/77915, PR fortran/77942.

  [ Matthias Klose ]
  * Backport Mips go closure support, taken from libffi. Closes: #839132.
  * Configure with --enable-default-pie and pass -z now when pie is enabled;
    on amd64 arm64 armel armhf i386 mips mipsel mips64el ppc64el s390x.
    Closes: #835148.
  * Update the Linaro support to the 6-2016.10 snapshot.

  [ Aurelien Jarno ]
  * Enable logwatch on mips64el.

 -- Matthias Klose <doko@debian.org>  Tue, 18 Oct 2016 13:53:00 +0200
```

Debian Jessie:
```
debian@test-bbbw-11:~$ cat /etc/debian_version ; uname -r ; dmesg | grep uio ; lsmod
8.6
4.4.30-ti-rt-r64
Module                  Size  Used by
ccm                     8777  3 
arc4                    2211  2 
wl18xx                101164  0 
wlcore                229104  1 wl18xx
bnep                   16269  2 
mac80211              654881  2 wl18xx,wlcore
cfg80211              555914  3 mac80211,wl18xx,wlcore
snd_soc_hdmi_codec      6824  1 
hci_uart               64496  1 
btbcm                   8044  1 hci_uart
btqca                   8077  1 hci_uart
bluetooth             521023  23 bnep,btbcm,btqca,hci_uart
rfkill                 21058  6 cfg80211,bluetooth
snd_soc_simple_card     9804  0 
pvrsrvkm              438900  0 
omap_sham              26611  0 
omap_aes_driver        24010  0 
snd_soc_davinci_mcasp    20606  2 
snd_soc_edma            1290  1 snd_soc_davinci_mcasp
snd_soc_omap            3762  1 snd_soc_davinci_mcasp
omap_rng                5544  0 
rng_core                9042  1 omap_rng
tilcdc                 31733  0 
wlcore_sdio             8781  0 
snd_soc_core          191839  5 snd_soc_hdmi_codec,snd_soc_davinci_mcasp,snd_soc_edma,snd_soc_omap,snd_soc_simple_card
snd_pcm_dmaengine       5894  2 snd_soc_core,snd_soc_omap
snd_pcm               103302  5 snd_soc_hdmi_codec,snd_soc_davinci_mcasp,snd_soc_core,snd_soc_omap,snd_pcm_dmaengine
snd_timer              23562  1 snd_pcm
snd                    72190  3 snd_soc_core,snd_timer,snd_pcm
soundcore               8296  1 snd
evdev                  13460  1 
tda998x                15790  0 
uio_pdrv_genirq         4045  0 
uio                    10497  1 uio_pdrv_genirq
usb_f_acm               8345  1 
u_serial               13119  3 usb_f_acm
usb_f_rndis            25919  1 
g_multi                 6010  0 
usb_f_mass_storage     49445  2 g_multi
u_ether                14395  2 usb_f_rndis,g_multi
libcomposite           53799  4 usb_f_acm,usb_f_rndis,g_multi,usb_f_mass_storage
debian@test-bbbw-11:~$
```
Debian Stretch:
```
debian@test-bbb-2:~$ cat /etc/debian_version ; uname -r ; dmesg | grep uio ; lsmod
stretch/sid
4.4.30-ti-r64
[   11.444001] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.448588] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.449681] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.473684] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.490112] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.516298] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.633743] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.645384] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.693955] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.717017] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   11.733640] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.024335] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.073147] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.078842] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.108182] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.109899] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.635029] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.635888] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.637172] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.637994] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.641682] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.655608] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.667670] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.704757] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.709557] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.719808] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.721290] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.722576] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.728144] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.762803] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.766332] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.790212] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.814894] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.842438] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.909659] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.925343] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.929345] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   12.930608] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.306444] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.318093] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.330661] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.355788] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.380692] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.386883] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.413377] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.414240] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.414879] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.417647] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.434316] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.457322] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.474584] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.478358] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.499692] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.506250] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.593276] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.594171] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.602691] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.629119] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.630983] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.633388] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
[   13.695493] uio: Unknown symbol _GLOBAL_OFFSET_TABLE_ (err 0)
Module                  Size  Used by
debian@test-bbb-2:~$
```
Regards,

**Comment by lromor** [comment]
@RobertCNelson Thanks!
